### PR TITLE
Release/0.8.0

### PR DESCRIPTION
**Pull request by rousso**
Review and merge this one to main. After releasing EXT 1.3.0 we will also release ENV 0.8.0.

### CODE REVIEW

**Comment by bertrand-lorentz** [comment]
If I run the application to convert an XML notice, I get the following error:
```
12:15:28.970 [main] DEBUG e.e.t.e.v.generator.HtmlGenerator - Applying XSL transformation for language [en] and view ID [CEI]
12:15:29.126 [main] ERROR e.e.ted.eforms.viewer.Application - Error executing the application with arguments [[en, ../../eforms-sdk/examples/notices/CEI.xml]]. Please see the logs.
12:15:29.128 [main] DEBUG e.e.ted.eforms.viewer.Application - Exception thrown:
net.sf.saxon.trans.XPathException: URIs using protocol file are not permitted
	at net.sf.saxon.lib.DirectResourceResolver.resolve(DirectResourceResolver.java:59)
	at net.sf.saxon.lib.ResourceRequest.resolve(ResourceRequest.java:123)
	at net.sf.saxon.functions.DocumentFn.resolveURI(DocumentFn.java:418)
. . .
	at eu.europa.ted.eforms.viewer.generator.HtmlGenerator.applyXslTransformation(HtmlGenerator.java:212)
	at eu.europa.ted.eforms.viewer.generator.HtmlGenerator.generateFile(HtmlGenerator.java:109)
	at eu.europa.ted.eforms.viewer.NoticeViewer.generateHtmlFile(NoticeViewer.java:123)
	at eu.europa.ted.eforms.viewer.NoticeViewer.generateHtmlFile(NoticeViewer.java:95)
	at eu.europa.ted.eforms.viewer.cli.CliCommand.call(CliCommand.java:112)
	at eu.europa.ted.eforms.viewer.cli.CliCommand.call(CliCommand.java:36)
. . .
```	
Unit tests run OK.

It seems that our custom URIResolver (TranslationUriResolver) is not used anymore when running the app from the command line.
I was not able to figure out how to use it without possibly impacting usage of the viewer as a library.

**Comment by bertrand-lorentz** [comment]
I added a commit to solve the problem noted above: running from the command line now also uses TranslationUriResolver.

This looks OK, but we should wait until eforms-toolkit-java is released before merging this and releasing.

**Review by bertrand-lorentz — DISMISSED** [review]

File: src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+   */
+  public static String computeKey(String... strings) {
+    Validate.notEmpty(strings, "The array of strings cannot be empty");
+
+    try {
+      String key = new String(
+          MessageDigest.getInstance("SHA-512")
+              .digest((StringUtils.join(strings, "###").getBytes())));
+
+      logger.trace("Computed key for [{}]: {}", strings, key);
```
Comment (rouschr) [inline src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java:129]: logger.trace("Computed key for [{}]: {}", strings, key);
It is not very clear where or what the key is in this message.
"Computed key [{}] for: {}", key, strings    would be simpler

File: src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+      return key;
+    } catch (NoSuchAlgorithmException e) {
+      throw new IllegalStateException("Failed to compute cache key", e);
```
Comment (rouschr) [inline src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java:133]: If this ever happens you would really want to know about the input strings, so use String.format("Failed to compute cache key for %s", strings); or something like that.

**Review by bertrand-lorentz — DISMISSED** [review]

File: src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java (hunk 2 above)
Comment (bertrand-lorentz) [reply src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java:133]: A NoSuchAlgorithmException would come from MessageDigest.getInstance("SHA-512"), so not related to the input strings.

**Review by bertrand-lorentz — APPROVED** [review]

File: src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java (hunk 1 above)
Comment (bertrand-lorentz) [reply src/main/java/eu/europa/ted/eforms/viewer/util/CacheHelper.java:129]: Current formulation is clearer for me.

**Pull request merged**